src/agent.py

```
@@ -259,59 +462,166 @@ async def fetch_long_term():
+    seen_content = set()
+    
+    for results in all_results:
+        for content, timestamp, metadata in results:
+            if content not in seen_content:
+                unique_observations.append((content, timestamp, metadata))
+                seen_content.add(content)
+    
+    return unique_observations
+
+
+def _format_observations(observations: list[tuple[str, str, dict]], include_premises: bool) -> str:
```

Pylint (3.3.7)

[refactor] Too many local variables (22/15)

(R0914)